storagemarket/provider.go

```diff
@@ -182,10 +180,12 @@ func (p *Provider) ExecuteDeal(dp *types.DealParams, clientPeer peer.ID) (pi *ap
 		Transfer:           dp.Transfer,
 	}
 	// validate the deal proposal
-	if err := p.validateDealProposal(ds); err != nil {
-		return &api.ProviderDealRejectionInfo{
-			Reason: fmt.Sprintf("failed validation: %s", err),
-		}, nil
+	if !p.testMode {
```

Comment:
If we added interfaces for PieceAdder and ChainDealManager, maybe add an interface for the validator as well?

Reply:
Hey yes, have added it as a TODO on the issue we have for writing tests. This will be in a follow-up PR.